BinaryObjectScanner/Protection/CopyX.cs

```diff
@@ -0,0 +1,171 @@
+    public class CopyX : IPathCheck, IPortableExecutableCheck
+    {
+        // https://web.archive.org/web/20011016234742/http://www.optimal-online.de:80/product/copy_x.htm
+        // There are four kinds of copy-X; Light, Profesisonal, audio, and Trial Maker.
+        // audio is for Audio CDs. Might be scannable, might not. Samples needed to confirm
+        // No samples of Trial are known at the moment, so it can't be checked for either.
+        // There are two kinds of copy-X generally observed; those with only rings, and those with rings and a disc check.
+        // These comments assume with 0 evidence that the former is Light and the latter is Professional.
+        // Because there is no evidence, only copy-X is being returned for now. This check has pre-emptively separated the two,
+        // just for when a designation can be applied for sure.
+
+        // Overall:
+        // Whenever these comments state "at the end of" or "at the start of" pertaining to the filesystem, they refer to alphabetical order, because this is how copy-X images seem to be mastered.
+        // Both Light and Professional have a directory at the end of the image. The files within this directory are intersected by the physical ring.
+        // This file is usually called ZDAT, but not always. At least one instance of Light calls it ZDATA. At least one instance of Professional calls it System.
+        // Seemingly it can be anything. It doesn't help that most known samples are specifically from one company's games, Tivola. Still, most use ZDAT.
+
+        // Professional:
+        // All instances of professional contain a disc check, performed via optgraph.dll. 
+        // All instances of professional contain in the directory at the end of the image 3 files. gov_[something].x64, iofile.x64, and sound.x64.
+        // Due to gov's minor name variance, sound.x64 sometimes being intersected by a ring at the start, and iofile.x64 being referenced directly in optgraph.x64, only iofile.x64 is being checked for now.
+        // TODO: optgraph.dll also contains DRM to prevent kernel debugger SoftICE from being used, via a process called SoftICE-Test. It is not currently known if this is specifically part of copy-X, or if it's an external solution employed by both copy-X and also other companies. If it's the latter, it should have its own check. It has none here since it wouldn't be necessary.
+
+
+        // Light:
+        // All instances of light contain 1 or more files in the directory at the end of the image. They all consist of either 0x00, or some data that matches between entries (and also is present in the 3 Professional files), except for the parts with the rings running through them.
+        // TODO: Check the last directory alphabetically and not just ZDAT*
+
+        /// <inheritdoc/>
+        public string? CheckPortableExecutable(string file, PortableExecutable pex, bool includeDebug)
+        {
```

Comment:
Take a quick look at other protections that implement `IPortableExecutableCheck`. There is a block that may appear useless where the section table is being checked and will return if it's missing. This is an intentional overly-paranoid check to ensure that other implementers of the library wouldn't run into odd issues.